spec/api/chemotion/reaction_api_spec.rb

```diff
@@ -0,0 +1,551 @@
+        expect(reactions.map { |r| [r['id'], r['name']] }).to match_array(
+          [[r1.id, r1.name], [r2.id, r2.name]]
+        )
+        expect(reactions.first).to include(
+          'id' => r2.id, 'name' => r2.name, 'type' => 'reaction',
+          'tag' => include(
+            'taggable_id' => r2.id, 'taggable_type' => 'Reaction',
+            'taggable_data' => include(
+              'collection_labels' => include(
+                'name' => 'C1', 'is_shared' => false, 'id' => c1.id,
+                'user_id' => user.id, 'shared_by_id' => c1.shared_by_id,
+                'is_synchronized' => c1.is_synchronized
+              )
+            )
+          )
+        )
+      end
+    end
+
+    context 'with ID of collection' do
+      before { get '/api/v1/reactions', params: { collection_id: c1.id } }
+
+      it 'returns serialized reaction' do
+        reactions = JSON.parse(response.body)['reactions']
+        expect(reactions.map { |reaction| reaction['id'] }).to eq([r2.id, r1.id])
+      end
+    end
+
+    context 'with ID of non-existing collection' do
+      before { get '/api/v1/reactions', params: { collection_id: 42 } }
+
+      it 'does not return reaction' do
+        reactions = JSON.parse(response.body)['reactions']
+        expect(reactions.size).to be(0)
+      end
+    end
+
+    context 'with ID of synced collection' do
+      let(:sharer) { create(:person) }
+      let(:sync_collections_user) do
+        create(:sync_collections_user, collection: c1, user: user, sharer: sharer)
+      end
+
+      before do
+        get '/api/v1/reactions', params: { sync_collection_id: sync_collections_user.id }
+      end
+
+      it 'returns serialized reaction' do
+        reactions = JSON.parse(response.body)['reactions']
+        expect(reactions.map { |reaction| reaction['id'] }).to eq([r2.id, r1.id])
+      end
+    end
+
+    context 'with ID of non-existing synced collection' do
+      before do
+        get '/api/v1/reactions', params: { sync_collection_id: 42 }
+      end
+
+      it 'does not return reaction' do
+        reactions = JSON.parse(response.body)['reactions']
+        expect(reactions.size).to be(0)
+      end
+    end
+  end
+
+  describe 'GET /api/v1/reactions/:id' do
+    context 'with appropriate permissions' do
+      let(:c1) do
+        create(:collection, user: user, is_shared: true, permission_level: 0)
+      end
+      let(:c2) { create(:collection, user: user) }
+      let(:r1) { create(:reaction) }
+      let(:r2) { create(:reaction) }
+
+      before do
+        CollectionsReaction.create!(collection_id: c1.id, reaction_id: r1.id)
+        CollectionsReaction.create!(collection_id: c2.id, reaction_id: r2.id)
+      end
+
+      describe 'reading reaction r1' do
+        before { get "/api/v1/reactions/#{r1.id}" }
+
+        it 'is allowed' do
+          expect(response.status).to eq 200
+        end
+      end
+
+      describe 'reading reaction r2' do
```

Comment:
Why testing the same behavior as in line 101?

Reply:
Right, seems redundant. See e71b7e2.